src/app/services/baw-api/baw-api.service.ts

```diff
@@ -441,19 +534,36 @@ export class BawApiService<
    *
    * @param path API path
    * @param body Request body
+   * @param headers Request headers
+   *
+   * @param options Options to modify the request
+   * eg. `{ withCredentials: false }` will not include the users authentication token and cookies in the request
    */
   public httpPatch(
     path: string,
     body?: any,
-    options: any = defaultApiHeaders
+    headers: HttpHeaders = defaultApiHeaders,
+    options: BawServiceOptions = defaultBawServiceOptions
   ): Observable<ApiResponse<Model>> {
+    const fullOptions = this.buildServiceOptions(options);
```

Comment:
looks like the options are all built twice, in the create, show, list... methods and in the lower level ones. Inefficient?

Reply:
I originally did this so that we wouldn't be relying on `undefined` as falsy values for `disableNotifications` eg. [in show](https://github.com/QutEcoacoustics/workbench-client/pull/2125/files/d4ce617e055ffa8add5b0a2c1e287104fbff2cd1#diff-2e5a0799ecfae901aaf8f95aa03ccce27de670e1faf96896372e2992493d13a3R263)

If the user passes in `{ cacheOptions: { cache: false } }`, `disableNotifications` will be `undefined` (defaulting to `false` because `undefined` is a falsy value).

This is currently how it works [proof](https://github.com/QutEcoacoustics/workbench-client/blob/master/src/app/services/baw-api/baw-api.service.ts#L179), but it might lead to some troubles down the road (as I believe that using `undefined` as a default "falsy" value is bad practice)

example of potential future pitfalls: if we change the `defaultOptions.disableNotifications` to `true` sometime in the future, it won't change some API calls where we set `withCredentials` or `cacheOptions`

---

In hindsight, we could do something like this

![image](https://github.com/QutEcoacoustics/workbench-client/assets/33742269/6ba2114e-92ab-4953-a27a-c4b5a568ddf0)

Where we make nullish `options?.disableNotifications` values default to the default options value